### ISSUE TITLE
macOS app crashes when creating GZipStream after change in code

### ISSUE THREAD

**Issue by tipa** [xamarin/xamarin-macios#10880]
### Steps to Reproduce

```
var bytes = new byte[] { 0x1, 0x2, 0x3, 0x3 };
var compressedStream = new MemoryStream();
var fileStream = new MemoryStream(bytes);
using (var compressionStream = new GZipStream(compressedStream, CompressionMode.Compress, true))
{
    fileStream.CopyTo(compressionStream);
}
var compressedBytes = compressedStream.ToArray();
```

- Build / run once
- Change some line of code
- Build / run again

### Expected Behavior
No crash

### Actual Behavior
App crashes when creating `new GZipStream()`
 macOS crash report: [Process: WorkingHours [45179].txt](https://github.com/xamarin/xamarin-macios/files/6147270/Process.WorkingHours.45179.txt)

### Additional comments
- using Mac mini with M1 chip
- problem does not occur when completely rebuilding before running after change (menu item "Rebuild App" instead of "Build App")
- also crashes when trying to decompress: `new GZipStream(compressedStream, CompressionMode.Decompress, true)`
- seems to be somehow related to libMonoPosixHelper and failing to resolve a dependency (System.IO.Compression)

### Environment

```
=== Visual Studio Community 2019 for Mac ===

Version 8.9.1 (build 34)
Installation UUID: 164a0a3e-1f7d-4617-8f0e-a0595e821464
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)

	Package version: 612000122

=== Mono Framework MDK ===

Runtime:
	Mono 6.12.0.122 (2020-02/c621c35ffa0) (64-bit)
	Package version: 612000122

=== Roslyn (Language Service) ===

3.9.0-6.21152.10+c10f884b30737542ddd84ca889a4aad9281ce210

=== NuGet ===

Version: 5.8.0.6860

=== .NET Core SDK ===

SDK: /usr/local/share/dotnet/sdk/5.0.201/Sdks
SDK Versions:
	5.0.201
	3.1.407
MSBuild SDKs: /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Sdks

=== .NET Core Runtime ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	5.0.4
	3.1.13

=== .NET Core 3.1 SDK ===

SDK: 3.1.407

=== Xamarin.Profiler ===

Version: 1.6.15.68
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Xamarin Designer ===

Version: 16.9.0.316
Hash: bd2705417
Branch: remotes/origin/d16-9
Build date: 2021-02-24 00:16:08 UTC

=== Apple Developer Tools ===

Xcode 12.4 (17801)
Build 12D4e

=== Xamarin.Mac ===

Version: 7.8.2.5 (Visual Studio Community)
Hash: 3836759d4
Branch: d16-9
Build date: 2021-02-10 17:56:43-0500

=== Xamarin.iOS ===

Version: 14.14.2.5 (Visual Studio Community)
Hash: 3836759d4
Branch: d16-9
Build date: 2021-02-10 17:56:44-0500

=== Xamarin.Android ===

Not Installed

=== Microsoft OpenJDK for Mobile ===

Java SDK: Not Found

Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android SDK Manager ===

Version: 16.9.0.22
Hash: a391de2
Branch: remotes/origin/d16-9
Build date: 2021-02-18 03:14:56 UTC

=== Android Device Manager ===

Version: 16.9.0.17
Hash: fc2b3db
Branch: remotes/origin/d16-9
Build date: 2021-02-18 03:15:18 UTC

=== Build Information ===

Release ID: 809010034
Git revision: 33bce2b7f3d45d05865cd3282f1c2cdc3a112aa0
Build date: 2021-03-04 11:21:38-05
Build branch: release-8.9
Xamarin extensions: 33bce2b7f3d45d05865cd3282f1c2cdc3a112aa0

=== Operating System ===

Mac OS X 10.16.0
Darwin 20.3.0 Darwin Kernel Version 20.3.0
    Thu Jan 21 00:06:51 PST 2021
    root:xnu-7195.81.3~1/RELEASE_ARM64_T8101 x86_64

```

### Build Logs

```

Building WorkingHours.MacOS (Debug)
Build started 3/16/2021 9:21:26 AM.
__________________________________________________
Project "/Users/timopartl/GitHub/workinghours/WorkingHours.MacOS/WorkingHours.MacOS.csproj" (Build target(s)):

Target _BeforeCoreCompileImageAssets:
  Skipping target "_BeforeCoreCompileImageAssets" because all output files are up-to-date with respect to the input files.
Target _CoreCompileImageAssets:
  Skipping target "_CoreCompileImageAssets" because all output files are up-to-date with respect to the input files.
Target _BeforeCoreCompileInterfaceDefinitions:
  Skipping target "_BeforeCoreCompileInterfaceDefinitions" because all output files are up-to-date with respect to the input files.
Target _CoreCompileInterfaceDefinitions:
  Skipping target "_CoreCompileInterfaceDefinitions" because all output files are up-to-date with respect to the input files.
Target _BeforeCoreCompileSceneKitAssets:
  Skipping target "_BeforeCoreCompileSceneKitAssets" because it has no inputs.
Target _BeforeCoreCompileSceneKitAssets:
  Skipping target "_BeforeCoreCompileSceneKitAssets" because it has no inputs.
Target _BeforeCoreCompileSceneKitAssets:
  Skipping target "_BeforeCoreCompileSceneKitAssets" because it has no inputs.
Target _CoreCompileSceneKitAssets:
  Skipping target "_CoreCompileSceneKitAssets" because it has no inputs.
Target _CoreCompileColladaAssets:
  Skipping target "_CoreCompileColladaAssets" because it has no inputs.
Target _BeforeCompileCoreMLModels:
  Skipping target "_BeforeCompileCoreMLModels" because all output files are up-to-date with respect to the input files.
Target _CoreCompileCoreMLModels:
  Skipping target "_CoreCompileCoreMLModels" because all output files are up-to-date with respect to the input files.
Target ResolveAssemblyReferences:
    No way to resolve conflict between "netstandard, Version=2.1.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" and "netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51". Choosing "netstandard, Version=2.1.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" arbitrarily.
    No way to resolve conflict between "Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed" and "Newtonsoft.Json, Version=9.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed". Choosing "Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed" arbitrarily.
    No way to resolve conflict between "System.Runtime, Version=4.0.20.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" and "System.Runtime, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a". Choosing "System.Runtime, Version=4.0.20.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" arbitrarily.
    No way to resolve conflict between "System.Reflection, Version=4.0.10.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" and "System.Reflection, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a". Choosing "System.Reflection, Version=4.0.10.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" arbitrarily.
    No way to resolve conflict between "System.Collections, Version=4.0.10.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" and "System.Collections, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a". Choosing "System.Collections, Version=4.0.10.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" arbitrarily.
    No way to resolve conflict between "System.Runtime.Extensions, Version=4.0.10.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" and "System.Runtime.Extensions, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a". Choosing "System.Runtime.Extensions, Version=4.0.10.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" arbitrarily.
    Consider app.config remapping of assembly "Newtonsoft.Json, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed" from Version "11.0.0.0" [] to Version "12.0.0.0" [/Users/timopartl/.nuget/packages/newtonsoft.json/12.0.3/lib/netstandard2.0/Newtonsoft.Json.dll] to solve conflict and get rid of warning.
    Consider app.config remapping of assembly "System.Memory, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" from Version "4.0.1.0" [] to Version "4.0.1.1" [/Users/timopartl/.nuget/packages/system.memory/4.5.3/lib/netstandard2.0/System.Memory.dll] to solve conflict and get rid of warning.
    Consider app.config remapping of assembly "System.Runtime.CompilerServices.Unsafe, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" from Version "4.0.4.1" [] to Version "5.0.0.0" [/Users/timopartl/.nuget/packages/system.runtime.compilerservices.unsafe/5.0.0/ref/netstandard2.1/System.Runtime.CompilerServices.Unsafe.dll] to solve conflict and get rid of warning.
    /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Microsoft.Common.CurrentVersion.targets(2143,5): warning MSB3277: Found conflicts between different versions of "Microsoft.CSharp" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed.
    /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Microsoft.Common.CurrentVersion.targets(2143,5): warning MSB3277: Found conflicts between different versions of "System.Numerics.Vectors" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed.
    Consider app.config remapping of assembly "netstandard, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" from Version "2.0.0.0" [] to Version "2.1.0.0" [/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/netstandard.dll] to solve conflict and get rid of warning.
    Consider app.config remapping of assembly "System.Runtime, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" from Version "4.0.0.0" [] to Version "4.0.20.0" [] to solve conflict and get rid of warning.
    Consider app.config remapping of assembly "System.Reflection, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" from Version "4.0.0.0" [] to Version "4.0.10.0" [] to solve conflict and get rid of warning.
    Consider app.config remapping of assembly "System.Collections, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" from Version "4.0.0.0" [] to Version "4.0.10.0" [/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Collections.dll] to solve conflict and get rid of warning.
    Consider app.config remapping of assembly "System.Runtime.Extensions, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" from Version "4.0.0.0" [] to Version "4.0.10.0" [] to solve conflict and get rid of warning.
    /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Microsoft.Common.CurrentVersion.targets(2143,5): warning MSB3276: Found conflicts between different versions of the same dependent assembly. Please set the "AutoGenerateBindingRedirects" property to true in the project file. For more information, see http://go.microsoft.com/fwlink/?LinkId=294190.
Done building target "ResolveAssemblyReferences" in project "WorkingHours.MacOS.csproj".
Target GenerateTargetFrameworkMonikerAttribute:
  Skipping target "GenerateTargetFrameworkMonikerAttribute" because all output files are up-to-date with respect to the input files.
Target XamlG:
  Skipping target "XamlG" because it has no outputs.
Target CssG:
  Skipping target "CssG" because it has no outputs.
Target CssG:
  Skipping target "CssG" because it has no outputs.
Target XamlG:
  Skipping target "XamlG" because it has no outputs.
Target CoreCompile:
    /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Roslyn/csc.exe /noconfig /nowarn:1701,1702,2008 /fullpaths /nostdlib+ /errorreport:prompt /warn:4 /define:__MACOS__;__UNIFIED__;DEBUG /errorendlocation /preferreduilang:en-US /reference:/Users/timopartl/.nuget/packages/dropbox.api/6.1.0/lib/netstandard2.0/Dropbox.Api.dll /reference:/Users/timopartl/.nuget/packages/google.apis.auth/1.50.0/lib/netstandard2.0/Google.Apis.Auth.dll /reference:/Users/timopartl/.nuget/packages/google.apis.auth/1.50.0/lib/netstandard2.0/Google.Apis.Auth.PlatformServices.dll /reference:/Users/timopartl/.nuget/packages/google.apis.core/1.50.0/lib/netstandard2.0/Google.Apis.Core.dll /reference:/Users/timopartl/.nuget/packages/google.apis/1.50.0/lib/netstandard2.0/Google.Apis.dll /reference:/Users/timopartl/.nuget/packages/google.apis.drive.v3/1.50.0.2250/lib/netstandard2.0/Google.Apis.Drive.v3.dll /reference:/Users/timopartl/.nuget/packages/guerrillantp/1.4.2/lib/netstandard2.0/GuerrillaNtp.dll /reference:/Users/timopartl/.nuget/packages/microsoft.bcl.asyncinterfaces/5.0.0/lib/netstandard2.1/Microsoft.Bcl.AsyncInterfaces.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Microsoft.CSharp.dll /reference:/Users/timopartl/.nuget/packages/microsoft.graph.core/1.24.0/lib/xamarinmac20/Microsoft.Graph.Core.dll /reference:/Users/timopartl/.nuget/packages/microsoft.graph/3.27.0/lib/netstandard2.0/Microsoft.Graph.dll /reference:/Users/timopartl/.nuget/packages/microsoft.identity.client/4.27.0/lib/xamarinmac20/Microsoft.Identity.Client.dll /reference:/Users/timopartl/.nuget/packages/microsoft.identity.client.extensions.msal/2.18.0/lib/netstandard2.0/Microsoft.Identity.Client.Extensions.Msal.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/mscorlib.dll /reference:/Users/timopartl/.nuget/packages/newtonsoft.json/12.0.3/lib/netstandard2.0/Newtonsoft.Json.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/OpenTK.dll /reference:/Users/timopartl/.nuget/packages/sentry/3.1.0/lib/netstandard2.1/Sentry.dll /reference:/Users/timopartl/.nuget/packages/simplexcel/3.0.2.193/lib/netstandard2.1/Simplexcel.dll /reference:/Users/timopartl/.nuget/packages/sqlite-net-pcl/1.8.0-beta/lib/netstandard2.0/SQLite-net.dll /reference:/Users/timopartl/.nuget/packages/sqlitepclraw.bundle_green/2.0.4/lib/netstandard2.0/SQLitePCLRaw.batteries_v2.dll /reference:/Users/timopartl/.nuget/packages/sqlitepclraw.core/2.0.4/lib/netstandard2.0/SQLitePCLRaw.core.dll /reference:/Users/timopartl/.nuget/packages/sqlitepclraw.provider.e_sqlite3/2.0.4/lib/netstandard2.0/SQLitePCLRaw.provider.e_sqlite3.dll /reference:/Users/timopartl/.nuget/packages/syncfusion.xamarin.core/18.4.0.47/lib/Xamarin.Mac/Syncfusion.Core.XForms.dll /reference:/Users/timopartl/.nuget/packages/syncfusion.xamarin.core/18.4.0.47/lib/Xamarin.Mac/Syncfusion.Core.XForms.macOS.dll /reference:/Users/timopartl/.nuget/packages/syncfusion.licensing/18.4.0.47/lib/Xamarin.Mac/Syncfusion.Licensing.dll /reference:/Users/timopartl/.nuget/packages/syncfusion.xamarin.sfchart/18.4.0.47/lib/Xamarin.Mac/Syncfusion.SfChart.XForms.dll /reference:/Users/timopartl/.nuget/packages/syncfusion.xamarin.sfchart/18.4.0.47/lib/Xamarin.Mac/Syncfusion.SfChart.XForms.macOS.dll /reference:/Users/timopartl/.nuget/packages/system.buffers/4.5.1/ref/netstandard2.0/System.Buffers.dll /reference:/Users/timopartl/.nuget/packages/system.collections.immutable/5.0.0/lib/netstandard2.0/System.Collections.Immutable.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/System.Core.dll /reference:/Users/timopartl/.nuget/packages/system.diagnostics.diagnosticsource/4.7.1/lib/netstandard1.3/System.Diagnostics.DiagnosticSource.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/System.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Drawing.Common.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/System.IdentityModel.dll /reference:/Users/timopartl/.nuget/packages/system.memory/4.5.3/lib/netstandard2.0/System.Memory.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/System.Net.Http.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/System.Numerics.Vectors.dll /reference:/Users/timopartl/.nuget/packages/system.reflection.metadata/5.0.0/lib/netstandard2.0/System.Reflection.Metadata.dll /reference:/Users/timopartl/.nuget/packages/system.runtime.compilerservices.unsafe/5.0.0/ref/netstandard2.1/System.Runtime.CompilerServices.Unsafe.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/System.Runtime.Serialization.dll /reference:/Users/timopartl/.nuget/packages/system.text.encodings.web/5.0.0/lib/netstandard2.1/System.Text.Encodings.Web.dll /reference:/Users/timopartl/.nuget/packages/system.text.json/5.0.0/lib/netstandard2.0/System.Text.Json.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/System.Xml.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/System.Xml.Linq.dll /reference:/Users/timopartl/.nuget/packages/webdav.client/2.7.0/lib/netstandard2.0/WebDav.Client.dll /reference:/Users/timopartl/.nuget/packages/xamarin.essentials/1.6.1/lib/xamarinmac20/Xamarin.Essentials.dll /reference:/Users/timopartl/.nuget/packages/xamarin.forms/4.5.0.617/lib/Xamarin.Mac/Xamarin.Forms.Core.dll /reference:/Users/timopartl/.nuget/packages/xamarin.forms/4.5.0.617/lib/Xamarin.Mac/Xamarin.Forms.Platform.dll /reference:/Users/timopartl/.nuget/packages/xamarin.forms/4.5.0.617/buildTransitive/XCODE11/Xamarin.Forms.Platform.macOS.dll /reference:/Users/timopartl/.nuget/packages/xamarin.forms/4.5.0.617/lib/Xamarin.Mac/Xamarin.Forms.Xaml.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Xamarin.Mac.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/Microsoft.Win32.Primitives.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/Microsoft.Win32.Registry.AccessControl.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/Microsoft.Win32.Registry.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/netstandard.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.AppContext.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Collections.Concurrent.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Collections.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Collections.NonGeneric.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Collections.Specialized.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.ComponentModel.Annotations.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.ComponentModel.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.ComponentModel.EventBasedAsync.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.ComponentModel.Primitives.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.ComponentModel.TypeConverter.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Console.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Data.Common.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Data.SqlClient.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Diagnostics.Contracts.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Diagnostics.Debug.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Diagnostics.FileVersionInfo.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Diagnostics.Process.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Diagnostics.StackTrace.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Diagnostics.TextWriterTraceListener.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Diagnostics.Tools.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Diagnostics.TraceEvent.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Diagnostics.TraceSource.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Diagnostics.Tracing.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Drawing.Primitives.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Dynamic.Runtime.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Globalization.Calendars.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Globalization.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Globalization.Extensions.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.IO.Compression.ZipFile.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.IO.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.IO.FileSystem.AccessControl.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.IO.FileSystem.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.IO.FileSystem.DriveInfo.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.IO.FileSystem.Primitives.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.IO.FileSystem.Watcher.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.IO.IsolatedStorage.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.IO.MemoryMappedFiles.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.IO.Pipes.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.IO.UnmanagedMemoryStream.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Linq.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Linq.Expressions.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Linq.Parallel.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Linq.Queryable.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Net.AuthenticationManager.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Net.Cache.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Net.HttpListener.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Net.Mail.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Net.NameResolution.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Net.NetworkInformation.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Net.Ping.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Net.Primitives.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Net.Requests.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Net.Security.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Net.ServicePoint.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Net.Sockets.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Net.Utilities.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Net.WebHeaderCollection.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Net.WebSockets.Client.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Net.WebSockets.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.ObjectModel.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Reflection.DispatchProxy.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Reflection.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Reflection.Emit.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Reflection.Emit.ILGeneration.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Reflection.Emit.Lightweight.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Reflection.Extensions.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Reflection.Primitives.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Reflection.TypeExtensions.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Resources.Reader.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Resources.ReaderWriter.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Resources.ResourceManager.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Resources.Writer.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Runtime.CompilerServices.VisualC.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Runtime.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Runtime.Extensions.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Runtime.Handles.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Runtime.InteropServices.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Runtime.InteropServices.RuntimeInformation.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Runtime.InteropServices.WindowsRuntime.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Runtime.Loader.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Runtime.Numerics.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Runtime.Serialization.Formatters.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Runtime.Serialization.Json.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Runtime.Serialization.Primitives.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Runtime.Serialization.Xml.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Security.AccessControl.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Security.Claims.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Security.Cryptography.Algorithms.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Security.Cryptography.Cng.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Security.Cryptography.Csp.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Security.Cryptography.DeriveBytes.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Security.Cryptography.Encoding.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Security.Cryptography.Encryption.Aes.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Security.Cryptography.Encryption.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Security.Cryptography.Encryption.ECDiffieHellman.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Security.Cryptography.Encryption.ECDsa.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Security.Cryptography.Hashing.Algorithms.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Security.Cryptography.Hashing.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Security.Cryptography.OpenSsl.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Security.Cryptography.Pkcs.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Security.Cryptography.Primitives.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Security.Cryptography.ProtectedData.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Security.Cryptography.RandomNumberGenerator.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Security.Cryptography.RSA.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Security.Cryptography.X509Certificates.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Security.Principal.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Security.Principal.Windows.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Security.SecureString.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.ServiceModel.Duplex.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.ServiceModel.Http.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.ServiceModel.NetTcp.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.ServiceModel.Primitives.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.ServiceModel.Security.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.ServiceProcess.ServiceController.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Text.Encoding.CodePages.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Text.Encoding.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Text.Encoding.Extensions.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Text.RegularExpressions.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Threading.AccessControl.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Threading.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Threading.Overlapped.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Threading.Tasks.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Threading.Tasks.Extensions.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Threading.Tasks.Parallel.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Threading.Thread.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Threading.ThreadPool.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Threading.Timer.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.ValueTuple.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Xml.ReaderWriter.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Xml.XDocument.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Xml.XmlDocument.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Xml.XmlSerializer.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Xml.XPath.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Xml.XPath.XDocument.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Xml.XPath.XmlDocument.dll /reference:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Facades/System.Xml.Xsl.Primitives.dll /debug+ /debug:full /nowin32manifest /optimize- /out:obj/Debug/WorkingHours.exe /target:exe /utf8output /langversion:9.0 AppDelegate.cs [...] "obj/Debug/Xamarin.Mac,Version=v2.0.AssemblyAttributes.cs"
    Using shared compilation with compiler from directory: /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Roslyn
    /Users/timopartl/GitHub/workinghours/WorkingHours/Sync/iCloudHelper.cs(251,26,251,27): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' annotations context.
Done building target "CoreCompile" in project "WorkingHours.MacOS.csproj".
Target XamlC:
    Compiling Xaml, assembly: obj/Debug/WorkingHours.exe
    Touching "obj/Debug/XamlC.stamp".
Target _CopyFilesMarkedCopyLocal:
    Touching "/Users/timopartl/GitHub/workinghours/WorkingHours.MacOS/obj/Debug/WorkingHours.MacOS.csproj.CopyComplete".
Target CopyFilesToOutputDirectory:
    Copying file from "/Users/timopartl/GitHub/workinghours/WorkingHours.MacOS/obj/Debug/WorkingHours.exe" to "/Users/timopartl/GitHub/workinghours/WorkingHours.MacOS/bin/Debug/WorkingHours.exe".
    WorkingHours.MacOS -> /Users/timopartl/GitHub/workinghours/WorkingHours.MacOS/bin/Debug/WorkingHours.exe
    Copying file from "/Users/timopartl/GitHub/workinghours/WorkingHours.MacOS/obj/Debug/WorkingHours.pdb" to "/Users/timopartl/GitHub/workinghours/WorkingHours.MacOS/bin/Debug/WorkingHours.pdb".
Target _DetectSigningIdentity:
    Detected signing identity:
            
      Code Signing Key: "Apple Development: XXXX (XXX)" (XXX)
      Provisioning Profile: "WorkingHours MacOS Dev" (XXX)
      Bundle Id: mac.partl.workinghours
      App Id: P9337B87X7.mac.partl.workinghours
Target _CopyResourcesToBundle:
  Skipping target "_CopyResourcesToBundle" because all output files are up-to-date with respect to the input files.
Target _CompileAppManifest:
  Skipping target "_CompileAppManifest" because all output files are up-to-date with respect to the input files.
Target _CompileToNative:
    /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/bin/mmp @/Users/timopartl/GitHub/workinghours/WorkingHours.MacOS/obj/Debug/response-file.rsp --xml=LinkDescription.xml 
    MMP : warning MM2006: Native library 'libsecret-1.so.0' was referenced but could not be found.
    		
    Touching "bin/Debug/bundler.stamp".
Done building target "_CompileToNative" in project "WorkingHours.MacOS.csproj".
Target _CopyAppExtensionsToBundle:
  Skipping target "_CopyAppExtensionsToBundle" because it has no inputs.
Target _CodesignNativeLibraries:
    Tool /usr/bin/codesign execution started with arguments: -v --force --timestamp --sign XXXX --timestamp=none /Users/timopartl/GitHub/workinghours/WorkingHours.MacOS/bin/Debug/WorkingHours.app/Contents/MonoBundle/libe_sqlite3.dylib
            
Target _CollectFrameworks:
    No Frameworks directory found.
            
Target _CodesignAppBundle:
    Tool /usr/bin/codesign execution started with arguments: -v --force -o runtime --timestamp --sign XXXX --entitlements /Users/timopartl/GitHub/workinghours/WorkingHours.MacOS/obj/Debug/Entitlements.xcent /Users/timopartl/GitHub/workinghours/WorkingHours.MacOS/bin/Debug/WorkingHours.app
            
Target _CodesignVerify:
    /usr/bin/codesign --verify -vvvv --deep bin/Debug/WorkingHours.app 
    bin/Debug/WorkingHours.app: valid on disk
    bin/Debug/WorkingHours.app: satisfies its Designated Requirement

Done building project "WorkingHours.MacOS.csproj".

========== Build: 1 succeeded, 0 failed, 2 up-to-date, 0 skipped ==========

Build: 0 errors, 5 warnings

Build succeeded.

/Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Microsoft.Common.CurrentVersion.targets(2143,5): warning MSB3277: Found conflicts between different versions of "Microsoft.CSharp" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed.
/Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Microsoft.Common.CurrentVersion.targets(2143,5): warning MSB3277: Found conflicts between different versions of "System.Numerics.Vectors" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed.
/Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Microsoft.Common.CurrentVersion.targets(2143,5): warning MSB3276: Found conflicts between different versions of the same dependent assembly. Please set the "AutoGenerateBindingRedirects" property to true in the project file. For more information, see http://go.microsoft.com/fwlink/?LinkId=294190.
/Users/timopartl/GitHub/workinghours/WorkingHours/Sync/iCloudHelper.cs(251,26,251,27): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' annotations context.
MMP : warning MM2006: Native library 'libsecret-1.so.0' was referenced but could not be found.
    5 Warning(s)
    0 Error(s)

Time Elapsed 00:00:09.03
```

### Example Project (If Possible)
I tried to create a sample project, however it was not reproducible with a demo project.
It does occur on both of my macOS Apps however, which are using System.IO.Compression

**Comment by tipa** [comment]
Today I got the same or a similar crash that didn't cause the app to close but the debugger stopped and I was able to inspect the exception:
![Screenshot 2021-03-18 at 07 46 39](https://user-images.githubusercontent.com/2863198/111584397-145b5d00-87be-11eb-87c1-53498283d713.png)

```
  at (wrapper managed-to-native) System.IO.Compression.DeflateStreamNative.CreateZStream(System.IO.Compression.CompressionMode,bool,System.IO.Compression.DeflateStreamNative/UnmanagedReadOrWrite,intptr)
  at System.IO.Compression.DeflateStreamNative.Create (System.IO.Stream compressedStream, System.IO.Compression.CompressionMode mode, System.Boolean gzip) [0x00036] in /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/src/Xamarin.Mac/mcs/class/System/System.IO.Compression/DeflateStream.cs:371 
  at System.IO.Compression.DeflateStream..ctor (System.IO.Stream compressedStream, System.IO.Compression.CompressionMode mode, System.Boolean leaveOpen, System.Boolean gzip) [0x0002d] in /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/src/Xamarin.Mac/mcs/class/System/System.IO.Compression/DeflateStream.cs:79 
  at System.IO.Compression.DeflateStream..ctor (System.IO.Stream stream, System.IO.Compression.CompressionMode mode, System.Boolean leaveOpen, System.Int32 windowsBits) [0x00000] in /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/src/Xamarin.Mac/mcs/class/System/System.IO.Compression/DeflateStream.cs:65 
  at (wrapper remoting-invoke-with-check) System.IO.Compression.DeflateStream..ctor(System.IO.Stream,System.IO.Compression.CompressionMode,bool,int)
  at System.IO.Compression.GZipStream..ctor (System.IO.Stream stream, System.IO.Compression.CompressionMode mode, System.Boolean leaveOpen) [0x00006] in /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/src/Xamarin.Mac/external/corefx/src/System.IO.Compression/src/System/IO/Compression/GZipStream.cs:21 
  at (wrapper remoting-invoke-with-check) System.IO.Compression.GZipStream..ctor(System.IO.Stream,System.IO.Compression.CompressionMode,bool)
  at WorkingHours.DatabaseHelper.ApplyNewDatabase (System.IO.Stream compressedStream, System.Boolean merge, System.Boolean forceReplace) [0x00010] in ... 
  at WorkingHours.OneDriveHelper.Sync (WorkingHours.SyncFeedbackHost feedbackHost, WorkingHours.SyncAction syncAction, System.Boolean merge, WorkingHours.OneDriveHelper+OneDriveSyncStatus syncStatus, System.Boolean isRetry) [0x00c39] in ... 
```

**Comment by mandel-macaque** [comment]
hmm looks like a mono issue. I wrote the bindings from the Compression API from Apple and I believe those should provide you with the same behaviour:  https://github.com/xamarin/xamarin-macios/blob/main/src/Compression/Compression.cs

Please check if that unblocks you, I'll track this done to mono.

**Comment by tipa** [comment]
Unfortunately, that doesn't help because the GZIP (de) compression isn't supported by this class.
Also, I am also using the `ZipArchive` class from System.IO.Compression

**Comment by mandel-macaque** [comment]
I have not been able to reproduce it on my side. Can you please do me a favour and check your system for the presence of DllNotFoundException: libMonoPosixHelper.dylib. 

Some small quesitons:

1. Are you using the system mono version?
2. How did you install the mono framework you are suing? Was it via our pkg of homebrew?

The Unity team hand an issue with this same library missing in the pkgs, could it be that you have Unity installed or that you have a bad installation of mono?

**Comment by tipa** [comment]
> Can you please do me a favour and check your system for the presence of DllNotFoundException: libMonoPosixHelper.dylib.

How exactly can I do that?

> * Are you using the system mono version?

What is the "system mono version"? How can I check that?

> 2\. How did you install the mono framework you are suing? Was it via our pkg of homebrew?

I usually only install Mono with the Visual Studio Updater. But I cannot completely rule out the possibility that I have installed another (beta) Mono version a few months ago. However, the reported issue started to show after the last big VS update. After I installed that VS update, I did not manually update Mono.

Maybe there is an option to send you my project confidentially?

**Comment by tipa** [comment]
Friendly ping @mandel-macaque :)
Was able to reproduce the same issue on a different device (Macbook Air with M1 CPU) - I just set up that device last week and I am sure I have not installed any custom Mono framework. The whole installation was done with the Visual Studio installer.

**Comment by spouliot** [comment]
>> Can you please do me a favour and check your system for the presence of DllNotFoundException: libMonoPosixHelper.dylib.

> How exactly can I do that?

Look for the file `libMonoPosixHelper.dylib` everywhere on your Mac. You might have several of them. Add the list (of them) on the issue.

E.g. Use `find` from the terminal

```bash
$ find / -name libMonoPosixHelper.dylib
...
/Library/Frameworks/Mono.framework//Versions/6.12.0/lib/libMonoPosixHelper.dylib
/Library/Frameworks/Mono.framework//Versions/6.12.0/lib/libMonoPosixHelper.dylib.dSYM/Contents/Resources/DWARF/libMonoPosixHelper.dylib
...
```

It shows I have the file inside my mono installation.


> What is the "system mono version"? How can I check that?

There's 3 different ways to build Xamarin.Mac applications.

1. Use the mobile profile. This is closer to Xamarin.iOS/tvOS/watchOS
2. Use the "full" profile. This is closer to the desktop FX, it allows more things and can ease compatibility with existing assemblies (that expect versions to be the one from the desktop FX)
3. Use the system mono. This means your app can only work if mono is installed. This is not supported (by us) as we have no control over the BCL being used. Some software uses this (like VSfM) because they enforce (and test) specific versions of mono to be installed.

Depending on your project options you'll be using one of the above methods. Failure to load a dll or native code can be quite different in each case.

**Comment by tipa** [comment]
```
/Library/Frameworks/Xamarin.Mac.framework/Versions/7.8.2.5/SDKs/Xamarin.macOS.sdk/lib/libMonoPosixHelper.dylib
/Library/Frameworks/Xamarin.Mac.framework/Versions/7.10.0.5/SDKs/Xamarin.macOS.sdk/lib/libMonoPosixHelper.dylib
/Library/Frameworks/Mono.framework/Versions/6.12.0/lib/libMonoPosixHelper.dylib
/Library/Frameworks/Mono.framework/Versions/6.12.0/lib/libMonoPosixHelper.dylib.dSYM/Contents/Resources/DWARF/libMonoPosixHelper.dylib
/Library/Frameworks/Xamarin.iOS.framework/Versions/14.16.0.5/lib/mlaunch/mlaunch.app/Contents/MonoBundle/libMonoPosixHelper.dylib
/Library/Frameworks/ObjectiveSharpie.framework/Versions/3.5.41/libMonoPosixHelper.dylib
/System/Volumes/Data/Library/Frameworks/Xamarin.Mac.framework/Versions/7.8.2.5/SDKs/Xamarin.macOS.sdk/lib/libMonoPosixHelper.dylib
/System/Volumes/Data/Library/Frameworks/Xamarin.Mac.framework/Versions/7.10.0.5/SDKs/Xamarin.macOS.sdk/lib/libMonoPosixHelper.dylib
/System/Volumes/Data/Library/Frameworks/Mono.framework/Versions/6.12.0/lib/libMonoPosixHelper.dylib
/System/Volumes/Data/Library/Frameworks/Mono.framework/Versions/6.12.0/lib/libMonoPosixHelper.dylib.dSYM/Contents/Resources/DWARF/libMonoPosixHelper.dylib
/System/Volumes/Data/Library/Frameworks/Xamarin.iOS.framework/Versions/14.16.0.5/lib/mlaunch/mlaunch.app/Contents/MonoBundle/libMonoPosixHelper.dylib
/System/Volumes/Data/Library/Frameworks/ObjectiveSharpie.framework/Versions/3.5.41/libMonoPosixHelper.dylib
/System/Volumes/Data/Applications/Appstat.app/Contents/MonoBundle/libMonoPosixHelper.dylib
/System/Volumes/Data/Applications/Visual Studio.app/Contents/MacOS/MonoDoc.app/Contents/MonoBundle/libMonoPosixHelper.dylib
/System/Volumes/Data/Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/AddIns/PerformanceDiagnostics/UIThreadMonitorDaemon.app/Contents/MonoBundle/libMonoPosixHelper.dylib
/System/Volumes/Data/Applications/Xamarin Profiler.app/Contents/MonoBundle/libMonoPosixHelper.dylib
/Applications/Visual Studio.app/Contents/MacOS/MonoDoc.app/Contents/MonoBundle/libMonoPosixHelper.dylib
/Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/AddIns/PerformanceDiagnostics/UIThreadMonitorDaemon.app/Contents/MonoBundle/libMonoPosixHelper.dylib
/Applications/Xamarin Profiler.app/Contents/MonoBundle/libMonoPosixHelper.dylib
```

These occurrences were found by the terminal command. It was also in the bin/obj folders of my apps, I excluded those lines.

I am using the project option below. I assume this indicates that I am using the "mobile profile"?
![Screenshot 2021-05-27 at 23 27 25](https://user-images.githubusercontent.com/2863198/119899225-23d7ee80-bf43-11eb-92dd-4172efe76efa.png)

**Comment by spouliot** [comment]
> /Library/Frameworks/Xamarin.Mac.framework/Versions/7.10.0.5/SDKs/Xamarin.macOS.sdk/lib/libMonoPosixHelper.dylib

I'm assuming you're using the latest, so 7.10.0.5, to build your app

If you run this command it will look at the file and tell you which architectures are available, it should look like

```
$ file /Library/Frameworks/Xamarin.Mac.framework/Versions/7.10.0.5/SDKs/Xamarin.macOS.sdk/lib/libMonoPosixHelper.dylib

/Library/Frameworks/Xamarin.Mac.framework/Versions/7.10.0.5/SDKs/Xamarin.macOS.sdk/lib/libMonoPosixHelper.dylib : Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit dynamically linked shared library x86_64] [arm64:Mach-O 64-bit dynamically linked shared library arm64]
/Library/Frameworks/Xamarin.Mac.framework/Versions/7.10.0.5/SDKs/Xamarin.macOS.sdk/lib/libMonoPosixHelper.dylib (for architecture x86_64):	Mach-O 64-bit dynamically linked shared library x86_64
/Library/Frameworks/Xamarin.Mac.framework/Versions/7.10.0.5/SDKs/Xamarin.macOS.sdk/lib/libMonoPosixHelper.dylib (for architecture arm64):	Mach-O 64-bit dynamically linked shared library arm64
```


>  I assume this indicates that I am using the "mobile profile"?

yes, you are

**Comment by tipa** [comment]
I am using VS 8.9.10 (build 4) and in the details it lists Xamarin.Mac 7.10.0.5 so I think this version is used to build my apps, yes.

The command you mentioned only prints one line for me:
```
/Library/Frameworks/Xamarin.Mac.framework/Versions/7.10.0.5/SDKs/Xamarin.macOS.sdk/lib/libMonoPosixHelper.dylib: Mach-O 64-bit dynamically linked shared library x86_64
```

**Comment by rolfbjarne** [comment]
My guess is that this is an issue where we don't resign libMonoPosixHelper.dylib for a rebuild, because the problem is that its code signature is wrong. From the crash report:

> Exception Type:        EXC_BAD_ACCESS (Code Signature Invalid)
> [...]
> Application Specific Information:
> dyld: in dlopen()
> /Users/timopartl/GitHub/workinghours/WorkingHours.MacOS/bin/Debug/WorkingHours.app/Contents/MonoBundle/libMonoPosixHelper.dylib

In these cases, a binlog from the build will usually make it possible for us to figure out what's going wrong.

How to get a binlog, adapted for your case:

* Build / run once
* Change some line of code
* Build / run again, but this time using the command line. Something like this:

```shell
# Build using 'msbuild', and pass the "/bl" flag, which instructs msbuild to create a binlog
msbuild /bl /Users/timopartl/GitHub/workinghours/WorkingHours.MacOS/WorkingHours.MacOS.csproj

# Run your app, and repro the problem (so that you're sure the rebuild wasn't done correctly)
open /Users/timopartl/GitHub/workinghours/WorkingHours.MacOS/bin/Debug/WorkingHours.app
```

After this is done, you'll have an `msbuild.binlog` file in the current directory. Zip this file up (because GitHub won't let you upload it otherwise), and attach it here.

Thanks!

**Comment by tipa** [comment]
Thanks for the clear instructions on how to create a binlog.
I did the procedure and attached the zipped binlog below:
[msbuild.binlog.zip](https://github.com/xamarin/xamarin-macios/files/6661183/msbuild.binlog.zip)

**Comment by rolfbjarne** [comment]
I think I know what's going on, it's a side effect of the fact that File.Copy preserves file stamps 😒 

1. App builds for the first time. 
    a. libMonoPosixHelper.dylib is copied to the app bundle
    b. We use a .stamp file to determine whether libMonoPosixHelper.dylib needs to be signed or not - this file does not exist, so we sign libMonoPosixHelper.dylib.
    c. We create the stamp file.
2. We build again.
    a. libMonoPosixHelper.dylib is once again copied to the app bundle. We use File.Copy, which means that we preserve the timestamp of the original libMonoPosixHelper.dylib.
    b. The code signing .stamp file for libMonoPosixHelper.dylib exists, so we check the timestamp. We see that the stamp file was created after libMonoPosixHelper.dylib, so we think that libMonoPosixHelper.dylib doesn't need to be signed.

**Comment by rolfbjarne** [comment]
Note: the actual fix is probably to not copy libMonoPosixHelper.dylib into the app the second time (by checking the timestamps of the existing version in the app we see that the newer version is the one in the app).

**Comment by tipa** [comment]
Any chance this could be fixed :)
I am creating `GZipStream`s right after app launch in my macOS apps.
Having to do a full rebuild every time makes the dev process reaaaally slow

**Comment by rolfbjarne** [comment]
This is a duplicate of #11862, so I'm closing this one.

There's a potential workaround here: https://github.com/xamarin/xamarin-macios/issues/11862#issuecomment-903638915.
